app/models/stage.rb

```diff
@@ -154,6 +156,14 @@ def send_github_notifications?
     update_github_pull_requests
   end
 
+  def production?
+    if ENV['DEPLOY_GROUP_FEATURE']
+      deploy_groups.empty? ? super : deploy_groups.any? { |deploy_group| deploy_group.environment.is_production? }
```

Comment:
I wonder if we could use `has_many :through` here so that you could do `stage.environments` directly